### PR TITLE
DLP: Added sample for Inspect phone number

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [inspect phone number](https://cloud.google.com/dlp/docs/samples/dlp-inspect-phone-number) with tests.

### CODE REVIEW

File: dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            Fixture = fixture;
+        }
+
+        [Fact]
+        public void TestInspectPhoneNumber()
+        {
+            var text = "My numbers are (415) 555-0890 and (415) 555-0990; and SSN is 372819127.";
+            var result = InspectPhoneNumber.Inspect(Fixture.ProjectId, text);
+            Assert.Equal(2, result.Result.Findings.Count);
```
Comment (annadata) [inline dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs:32]: Please also assert that the findings are for PHONE_NUMBER.

**Review by annadata — APPROVED** [review]

File: dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs (hunk 1 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs:34]: Done

File: dlp/api/Snippets/InspectPhoneNumber.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+// [START dlp_inspect_phone_number]
+
+using System;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Dlp.V2;
+
+public class InspectPhoneNumber
+{
+    public static InspectContentResponse Inspect(string projectId, string text, Likelihood minLikelihood = Likelihood.Possible)
```
Comment (amanda-tarafa) [inline dlp/api/Snippets/InspectPhoneNumber.cs:23]: Try to be consistent re styling, on the sample on https://github.com/GoogleCloudPlatform/dotnet-docs-samples/pull/2116 all the parameters are each on its own line, where here they are all on a single line.

We prefer all parameters in a single line unless that line will end up too long (>~80charaxters) in which case we put each parameter in a single line.

File: dlp/api/Snippets/InspectPhoneNumber.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            InfoTypes = { new InfoType { Name = "PHONE_NUMBER" } },
+            IncludeQuote = true,
+            MinLikelihood = minLikelihood
+        };
+
+        // Construct a request.
+        var request = new InspectContentRequest
+        {
+            Parent = new LocationName(projectId, "global").ToString(),
```
Comment (amanda-tarafa) [inline dlp/api/Snippets/InspectPhoneNumber.cs:42]: And please review all the other samples for this same change.

```suggestion
            ParentAsLocationName = new LocationName(projectId, "global"),
```

File: dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+using Xunit;
+
+namespace GoogleCloudSamples
+{
+    public class InspectPhoneNumbersTests : IClassFixture<DlpTestFixture>
+    {
+        private DlpTestFixture Fixture { get; }
```
Comment (amanda-tarafa) [inline dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs:21]: Same as in the other PR, prefer a private field here. Please review the rest of the PRs as well for this.

File: dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        public void TestInspectPhoneNumber()
+        {
+            var phoneNumber = "(415) 555-0890";
+            var ssn = "372819127";
+            var text = $"My numbers are {phoneNumber} and (415) 555-0990; and SSN is {ssn}.";
+            var result = InspectPhoneNumber.Inspect(Fixture.ProjectId, text);
+            Assert.Equal(2, result.Result.Findings.Count);
+            Assert.Contains(phoneNumber, result.ToString());
+            Assert.DoesNotContain(ssn, result.ToString());
+            Assert.Contains("PHONE_NUMBER", result.ToString());
```
Comment (amanda-tarafa) [inline dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs:37]: Something like this is better (might have syntax errors as I'm writing directly here, but you'll get the idea):
```suggestion
            Assert.Contains(result, f => f.InfoType.Name == "PHONE_NUMBER" && f.Quote == phoneNumber);
            Assert.DoesNotContain(result, f => f.Quote == ssn);
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Please review remaining PRs for the same suggestions as here.

Also, please rebase all changes into a single commit.

File: dlp/api/Snippets/InspectPhoneNumber.cs (hunk 2 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/InspectPhoneNumber.cs]: Added all the parameters on new line

File: dlp/api/Snippets/InspectPhoneNumber.cs (hunk 3 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/InspectPhoneNumber.cs]: Done

File: dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs (hunk 5 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/InspectPhoneNumberTests.cs]: Yes got it; And changed accordingly.

File: dlp/api/Snippets/InspectPhoneNumber.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            Item = contentItem,
+        };
+
+        // Call the API.
+        var response = dlp.InspectContent(request);
+
+        // Inspect the results.
+        var resultFindings = response.Result.Findings;
+
+        if (resultFindings.Count < 0)
```
Comment (amanda-tarafa) [inline dlp/api/Snippets/InspectPhoneNumber.cs:56]:
```suggestion
        if (resultFindings.Count <= 0)
```

**Review by amanda-tarafa — COMMENTED** [review]
A tiny change that I missed earlier. Otherwise LGTM.

File: dlp/api/Snippets/InspectPhoneNumber.cs (hunk 6 above)
Comment (amanda-tarafa) [reply dlp/api/Snippets/InspectPhoneNumber.cs:56]: Actually, on some of the other samples, the exemption is not made for when the results are empty. You just print the count and then execute the foreach which will do nothing if there are no resutls. Can we make this one similar to the others? Thanks

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]

File: dlp/api/Snippets/InspectPhoneNumber.cs (hunk 6 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/InspectPhoneNumber.cs]: removed the condition

**Review by amanda-tarafa — APPROVED** [review]
Thanks!

**Pull request merged**